src/Processors/Transforms/JoiningTransform.h

```diff
@@ -8,21 +8,63 @@ namespace DB
 class IJoin;
 using JoinPtr = std::shared_ptr<IJoin>;
 
-class JoiningTransform : public ISimpleTransform
+class IBlockInputStream;
+using BlockInputStreamPtr = std::shared_ptr<IBlockInputStream>;
+
+/// Join rows to chunk form left table.
+/// This transform usually has two input ports and one output.
+/// First input is for data from left table.
+/// Second input has empty header and is connected with AddingJoined.
+/// We can process left table only when Join is filled. Second input is used to signal that AddingJoined is finished.
```

Comment:
How does it handle switching to merge joins? For hash joins we first have to process the entire right side, but for merge joins, we process left and right sides simultaneously.

Reply:
For now, we have only "partial merge join". It also reads right table first.
Looks like now we can implement honest merge join